sway/desktop/xdg_shell.c

```diff
@@ -163,11 +179,17 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 	struct sway_xdg_shell_view *xdg_shell_view =
 		wl_container_of(listener, xdg_shell_view, commit);
 	struct sway_view *view = &xdg_shell_view->view;
-	// NOTE: We intentionally discard the view's desired width here
-	// TODO: Store this for restoration when moving to floating plane
-	// TODO: Let floating views do whatever
-	view_update_size(view, xdg_shell_view->pending_width,
-		xdg_shell_view->pending_height);
+	struct wlr_box *geometry = &view->wlr_xdg_surface->geometry;
```

Comment:
The xdg-surface geometry can be unset, in which case the surface's size should be used.

Reply:
Are you sure about that? `wlr_xdg_shell` contains these:

	bool has_next_geometry;
	struct wlr_box next_geometry;
	struct wlr_box geometry;

...which suggests to me that the geometry struct is always present.

Reply:
The client doesn't always set the geometry, so this box can be empty. We should probably document this, it's written in the protocol itself.